sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -288,8 +288,19 @@ private void CloseLink(ReceivingAmqpLink link)
             IEnumerable<string> lockTokens,
             TimeSpan timeout)
         {
-            Guid[] lockTokenGuids = lockTokens.Select(token => new Guid(token)).ToArray();
-            if (lockTokenGuids.Any(lockToken => _requestResponseLockedMessages.Contains(lockToken)))
+            bool requestResponse = false;
+            var lockTokenGuids = new List<Guid>();
+            foreach (string token in lockTokens)
+            {
+                var tokenGuid = new Guid(token);
+                lockTokenGuids.Add(tokenGuid);
+                if (!requestResponse && _requestResponseLockedMessages.Contains(tokenGuid))
```

Comment:
The idea is that we can avoid having to do the `Contains` if we already know we are in the request response scenario

Reply:
Why don't you break loop early in that case?

Reply:
Because my brain was still held captive in the previous version of the code. Stupid me. Will fix